rewrite-java/src/test/java/org/openrewrite/java/ChangeMethodParameterTest.java

```diff
@@ -0,0 +1,409 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class ChangeMethodParameterTest implements RewriteTest {
+
+    @DocumentExample
+    @Test
+    void primitive() {
+        rewriteRun(
+          spec -> spec.recipe(new ChangeMethodParameter("foo.Foo#bar(..)", "long", "j", 0)),
+          java(
+            """
+              package foo;
+              
+              public class Foo {
+                  public void bar(int i) {
+                  }
```

Comment:
I'm not so sure about changing the parameter name here, as we do not appear to change any usages. And even if we do change the usage, then there's quickly potential to break compilation, as seen in this example.
```suggestion
                  public void bar(int i) {
                      int plusOne = i + 1;
                  }
```
Right now the usage of `i` is not converted into `j`, and if it were, then the project would fail to compile. What are your thoughts here?

Reply:
Thanks for that context! I do indeed see the value in being able to express some initial migration steps in this way. I've also added a note to https://github.com/openrewrite/rewrite-spring/issues/265 for the Spring Batch migration step you pointed out.

Documentation would help in part to limit when to use the recipe; I think not offering the option to rename the parameter just yet might be easiest to start with.

There's also the option of only offering this as a visitor for now, and then have dedicated named recipes that use the visitor, knowing full well the limitations. That might prevent accidental incorrect use, but would also make it more cumbersome to define new steps in larger migrations that use this functionality.

Are you only planning this for Spring Batch for now? If so we could let it mature in rewrite-spring to start with, before pulling it up to rewrite-java at a later stage.

Reply:
@timtebeek Thanks again!

I agree that we can only allow to change parameter type and don't offering the option to rename the parameter, it's much safer.

And yes currently I'm planning for Spring Batch cases. For move to `rewrite-spring`, should I open a new PR at there?

Reply:
Yes that would work, thanks! I look forward to continuing the conversation there.

Reply:
Move to https://github.com/openrewrite/rewrite-spring/pull/631